src/components/common/SpectrumToggle.js

```diff
@@ -17,12 +21,33 @@ const useStyles = makeStyles(() => ({
 const SpectrumToggle = () => {
   const { t } = useTranslation();
   const spectrum = useSelector(({ lab }) => lab.spectrum);
+  const moleculesOnCanvas = useSelector(({ lab }) => lab.moleculesOnCanvas);
   const dispatch = useDispatch();
 
   const handleToggle = () => {
     if (spectrum === SPECTRUMS.VISIBLE_LIGHT) {
+      moleculesOnCanvas.forEach((molecule, index) => {
+        if (molecule.shouldOscillate) {
```

Comment:
(1) In almost every case using `.filter` with `moleculesOnCanvas` causes problems, because we lose track of the index a molecule is in. For example, say our original array has at index 0 a molecule with `shouldOscillate` `false` and at index 1 a molecule with `shouldOscillate` `true`. When we filter by `shouldOscillate`, in the filtered array the molecule originally at index 1 is now at index 0, meaning updating its state via its index causes problems.

(2) You're right - this code might be able to be simplified. When looking at it, I ended up finding a bug (#26). Will review this code with the bug.